src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -46,6 +49,11 @@
  */
 public final class BourneShellScript extends FileMonitoringTask {
 
+    /** Number of times we will show launch diagnostics in a newly encountered workspace before going mute to save resources. */
+    private static /* not final */ int NOVEL_WORKSPACE_DIAGNOSTICS_COUNT = Integer.getInteger(BourneShellScript.class.getName() + ".NOVEL_WORKSPACE_DIAGNOSTICS_COUNT", 10);
+    /** Number of seconds we will wait for a controller script to be launched before assuming the launch failed. */
+    private static /* not final */ int LAUNCH_FAILURE_TIMEOUT = Integer.getInteger(BourneShellScript.class.getName() + ".LAUNCH_FAILURE_TIMEOUT", 15);
```

Comment:
I do not expect these parameters to need to be configured; I am leaving the door open to experimentation without source code modification in case I receive bug reports or support requests that seem to be related.